packages/wdio-browserstack-service/src/ai-handler.ts

```diff
@@ -0,0 +1,212 @@
+import aiSDK from '@browserstack/ai-sdk-node'
+import { BStackLogger } from './bstackLogger.js'
+import { TCG_URL, TCG_INFO, SUPPORTED_BROWSERS_FOR_AI } from './constants.js'
+import { handleHealingInstrumentation } from './instrumentation/funnelInstrumentation.js'
+import { createRequire } from 'node:module'
+
+import type { Capabilities } from '@wdio/types'
+import type BrowserStackConfig from './config.js'
+import type { Options } from '@wdio/types'
+import type { BrowserstackHealing } from '@browserstack/ai-sdk-node'
+import path from 'node:path'
+import fs from 'node:fs'
+import url from 'node:url'
+import { getBrowserStackUserAndKey, isBrowserstackInfra } from './util.js'
+import type { BrowserstackOptions } from './types.js'
+
+class AiHandler {
+    authResult: BrowserstackHealing.InitSuccessResponse | BrowserstackHealing.InitErrorResponse
+    wdioBstackVersion: string
+    constructor() {
+        this.authResult = JSON.parse(process.env.TCG_AUTH_RESULT || '{}')
+        this.wdioBstackVersion = createRequire(import.meta.url)('../package.json').version
+    }
+
+    async authenticateUser(user: string, key: string) {
+        return await aiSDK.BrowserstackHealing.init(key, user, TCG_URL, this.wdioBstackVersion)
+    }
+
+    updateCaps(
+        authResult: BrowserstackHealing.InitSuccessResponse | BrowserstackHealing.InitErrorResponse,
+        options: BrowserstackOptions,
+        caps: Array<Capabilities.RemoteCapability> | Capabilities.RemoteCapability
+    ) {
+        const installExtCondition = authResult.isAuthenticated === true && (authResult.defaultLogDataEnabled === true || options.selfHeal === true)
+        if (installExtCondition){
+            if (Array.isArray(caps)) {
+                const newCaps= aiSDK.BrowserstackHealing.initializeCapabilities(caps[0])
+                caps[0] = newCaps
+            } else if (typeof caps === 'object') {
+                caps = aiSDK.BrowserstackHealing.initializeCapabilities(caps)
+            }
+        } else if (options.selfHeal === true) {
+            const healingWarnMessage = (authResult as aiSDK.BrowserstackHealing.InitErrorResponse).message
+            BStackLogger.warn(`Healing Auth failed. Disabling healing for this session. Reason: ${healingWarnMessage}`)
+        }
+
+        return caps
+    }
+
+    async setToken(sessionId: string, sessionToken: string){
+        await aiSDK.BrowserstackHealing.setToken(sessionId, sessionToken, TCG_URL)
+    }
+
+    async installFirefoxExtension(browser: WebdriverIO.Browser){
+        const __dirname =  url.fileURLToPath(new URL('.', import.meta.url))
+        const extensionPath = path.resolve(__dirname, aiSDK.BrowserstackHealing.getFirefoxAddonPath())
+        const extFile = fs.readFileSync(extensionPath)
+        await browser.installAddOn(extFile.toString('base64'), true)
```

Comment:
Don't we need to handle if path doesn't exist or not necessary permissions

Reply:
This is inside a try catch block (selfHeal) function so this should be fine